### PR TITLE
ARROW-17061: [Python][Substrait] Acero consumer is unable to consume count function from substrait query plan

### PR DESCRIPTION
**Pull request by vibhatha**
This PR is **WORK IN PROGRESS**. 

Testing the functions integrated to the Acero-Substrait via Python test cases.
Started possibly to test out a bug, but planning on re-iterating this into a function test suite.

### CODE REVIEW

File: python/pyarrow/tests/test_substrait.py [hunk 1]
```diff
@@ -165,3 +169,108 @@ def test_get_supported_functions():
...
+    file_name = "read_agg_data.arrow"
+    table = pa.Table.from_arrays([[1, 2, 3, 2, 3, 1, 1],
+                                  [10, 20, 30, 40, 50, 60, 70],
+                                  [3, 4, 5, 1, 2, 0, 20]],
+                                 names=['A', 'B', 'C'])
+    path = _write_dummy_data_to_disk(tmpdir, file_name, table)
+
+    query_with_path = substrait_query.replace("FILENAME_PLACEHOLDER", path)
+    query_with_ext_uri = query_with_path.replace(
+        "AGGREGATE_URI_PLACEHOLDER", _kAggregateURI)
```
Comment (richtia) [inline python/pyarrow/tests/test_substrait.py:268]: Are we going to need to do this for all of our substrait plans?  Shouldn't Acero be handling this part?

Comment (vibhatha) [reply python/pyarrow/tests/test_substrait.py:268]: This is part of the Substrait query, right? You would stil have to reference it AFAIK.

Comment (richtia) [reply python/pyarrow/tests/test_substrait.py:268]: The substrait plan is generated with the name of the yaml file in it already.  I think @westonpace is actually going to submit a PR to automatically reference the proper URI given the name of the yaml.

Comment (westonpace) [reply python/pyarrow/tests/test_substrait.py:268]: https://issues.apache.org/jira/browse/ARROW-17736 explains what I'm thinking.

Comment (vibhatha) [reply python/pyarrow/tests/test_substrait.py:268]: I will work on this one.

Comment (westonpace) [reply python/pyarrow/tests/test_substrait.py:268]: Ah.  I might keep this one actually.  I already have a PR ready to go from a few weeks back when I was playing around with this.  I just didn't have time to submit it, which I will do now.  I would appreciate a review / opinions on the change however.

**Comment by vibhatha** [comment]
This PR is no longer needed since this cause because of a format in the plan.

**Pull request closed without merging**